src/Raven.Client/ServerWide/Operations/UniqueValueOperation.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using Raven.Client.Documents.Conventions;
+using Raven.Client.Documents.Session;
+using Raven.Client.Http;
+using Raven.Client.Json;
+using Raven.Client.Util;
+using Sparrow.Json;
+
+namespace Raven.Client.ServerWide.Operations
+{
+    public class GetRawUniqueValueResult
```

Comment:
This is meant to _implement_ unique constraints, but it have far wider applications.
We shouldn't limit ourselves in the naming, use something like `cmpxchg` or `CompareExchange` instead